plugin/audit_log_v2/log_writer.cc

```diff
@@ -0,0 +1,138 @@
+/* Copyright (c) 2022 Percona LLC and/or its affiliates. All rights reserved.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA */
+
+#include "plugin/audit_log_v2/log_writer.h"
+
+#include "sql/mysqld.h"
+
+namespace audit_log_v2 {
+namespace {
+#if defined(HAVE_PSI_INTERFACE) && !defined(FLOGGER_NO_PSI)
+/* These belong to the service initialization */
+//  PSI_mutex_key key_LOCK_logger_service;
+//  PSI_mutex_info mutex_list[] = {{
+//    &key_LOCK_logger_service,
+//    "file_logger::lock",
+//    PSI_FLAG_SINGLETON,
+//    PSI_VOLATILITY_UNKNOWN,
+//    PSI_DOCUMENT_ME}};
+#else
+//#define key_LOCK_logger_service nullptr
+#endif /*HAVE_PSI_INTERFACE && !FLOGGER_NO_PSI*/
+
+}  // namespace
+
+LogWriterFile::LogWriterFile(LogWriterFileConfig conf)
+  : m_config{conf}, m_is_opened{false} {}
+
+LogWriterFile::~LogWriterFile() { do_close_file(); }
+
+bool LogWriterFile::open() { return do_open_file(); }
```

Comment:
To be honest I don't see a reason why we are not allowed to use exceptions in the new code.
In this case open() method can be called from the constructor.